jina/serve/runtimes/worker/__init__.py

```diff
@@ -224,6 +229,11 @@ async def endpoint_discovery(self, empty, context) -> jina_pb2.EndpointsProto:
         :param context: grpc context
         :returns: the response request
         """
+        if self._reject_incoming_requests:
```

Comment:
This is not to be handled by us. This is the health check job to do. And thus is why we set it as graceful shutdown

Reply:
My thought was that there is a brief moment where the worker is shutting down and processing pending requests, so the grpc server is still up and reachable, but should probably not accept new requests anymore.